cli/src/main/java/com/box/l10n/mojito/cli/command/PseudoLocCommand.java

```diff
@@ -0,0 +1,148 @@
+
+    CommandDirectories commandDirectories;
+
+    /**
+     * Contains a map of locale for generating localized file a locales defined
+     * in the repository.
+     */
+    Map<String, String> localeMappings;
+
+    @Override
+    public void execute() throws CommandException {
+        consoleWriter.newLine().a("Pull pseudo localized asset from repository: ").fg(Color.CYAN).a(repositoryParam).println(2);
+
+        repository = commandHelper.findRepositoryByName(repositoryParam);
+
+        commandDirectories = new CommandDirectories(sourceDirectoryParam, targetDirectoryParam);
+
+        for (FileMatch sourceFileMatch : commandHelper.getSourceFileMatches(commandDirectories, fileType, sourceLocale, sourcePathFilterRegex)) {
+            consoleWriter.a("Localizing: ").fg(Color.CYAN).a(sourceFileMatch.getSourcePath()).println();
+            generateLocalizedFiles(repository, sourceFileMatch);
+        }
+        consoleWriter.fg(Color.GREEN).newLine().a("Finished").println(2);
+    }
+
+    /**
+     * Default generation, uses the locales defined in the repository to
+     * generate the localized files.
+     *
+     * @param repository
+     * @param sourceFileMatch
+     * @throws CommandException
+     */
+    void generateLocalizedFiles(Repository repository, FileMatch sourceFileMatch) throws CommandException {
+        logger.debug("Generate pseudo localzied files");
+
+        LocalizedAssetBody localizedAsset = getLocalizedAsset(repository, sourceFileMatch);
+        writeLocalizedAssetToTargetDirectory(localizedAsset, sourceFileMatch);
+
+    }
+
+    void writeLocalizedAssetToTargetDirectory(LocalizedAssetBody localizedAsset, FileMatch sourceFileMatch) throws CommandException {
+        String outputBcp47tag = "en-x-psaccent";
+        localizedAsset.setBcp47Tag(outputBcp47tag);
+
+        Path targetPath = commandDirectories.getTargetDirectoryPath().resolve(sourceFileMatch.getTargetPath(localizedAsset.getBcp47Tag()));
+
+        commandHelper.writeFileContent(localizedAsset.getContent(), targetPath, sourceFileMatch);
+
+        Path relativeTargetFilePath = commandDirectories.relativizeWithUserDirectory(targetPath);
+
+        consoleWriter.a(" --> ").fg(Color.MAGENTA).a(relativeTargetFilePath.toString()).println();
+    }
+
+    LocalizedAssetBody getLocalizedAsset(Repository repository, FileMatch sourceFileMatch) throws CommandException {
+        consoleWriter.a(" - Processing locale: ").fg(Color.CYAN).a("en-usx").print();
+
+        try {
+            Asset assetByPathAndRepositoryId = assetClient.getAssetByPathAndRepositoryId(sourceFileMatch.getSourcePath(), repository.getId());
+
+            String assetContent = commandHelper.getFileContent(sourceFileMatch.getPath());
+
+            // TODO(P1) This is to inject xml:space="preserve" in the trans-unit element
+            // in the xcode-generated xliff until xcode fixes the bug of not adding this attribute
+            // See Xcode bug http://www.openradar.me/23410569
```

Comment:
There is a fair amount of duplicated code here from the pull command, this is not to great for maintenance. It would great to share the code